tests/testsuite/build_script_env.rs

```diff
@@ -171,17 +186,28 @@ fn rustc_bootstrap() {
         // NOTE: uses RUSTC_BOOTSTRAP so it will be propagated to rustc
         // (this matters when tests are being run with a beta or stable cargo)
         .env("RUSTC_BOOTSTRAP", "1")
-        .with_stderr_contains("warning: foo@0.0.1: Cannot set `RUSTC_BOOTSTRAP=1` [..]")
+        .with_stderr_data(str![[r#"
+[COMPILING] foo v0.0.1 ([ROOT]/foo)
+[WARNING] foo@0.0.1: Cannot set `RUSTC_BOOTSTRAP=1` from build script of `foo v0.0.1 ([ROOT]/foo)`.
+[NOTE] Crates cannot set `RUSTC_BOOTSTRAP` themselves, as doing so would subvert the stability guarantees of Rust for your project.
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+
+"#]])
         .run();
     // RUSTC_BOOTSTRAP conditionally set when there's no library should error (regardless of the value)
     p.cargo("check")
         .env("RUSTC_BOOTSTRAP", "foo")
-        .with_stderr_contains("error: Cannot set `RUSTC_BOOTSTRAP=1` [..]")
-        .with_stderr_contains("help: [..] set the environment variable `RUSTC_BOOTSTRAP=1` [..]")
+        .with_stderr_data(str![[r#"
+[ERROR] Cannot set `RUSTC_BOOTSTRAP=1` from build script of `foo v0.0.1 ([ROOT]/foo)`.
+[NOTE] Crates cannot set `RUSTC_BOOTSTRAP` themselves, as doing so would subvert the stability guarantees of Rust for your project.
+[HELP] If you're sure you want to do this in your project, set the environment variable `RUSTC_BOOTSTRAP=1` before running cargo instead.
+
+"#]])
         .with_status(101)
         .run();
 }
 
+#[allow(deprecated)]
```

Comment:
If the following test cases are all expanded, the content will be very large. I don't think of a better solution, maybe the current treatment is the most appropriate.

Reply:
Yes, for `-vv` output, we should probably still do "contains" checks as mentioned in the instructions

So
```rust
    p.cargo("check -vv")
        .with_stderr_contains("[WARNING] non-trivial mutual dependency between target-specific configuration and RUSTFLAGS")
        .with_stderr_contains("[foo 0.0.1] cfg!(bertrand)=true")
        .with_stderr_contains("[..]--cfg=bertrand[..]")
        .run();
```
could be
```rust
    p.cargo("check -vv")
        .with_stderr_contains("\
...
[WARNING] non-trivial mutual dependency between target-specific configuration and RUSTFLAGS
...")
        .with_stderr_contains("\
...
[foo 0.0.1] cfg!(bertrand)=true
...")
        .with_stderr_contains("\
...
[..]--cfg=bertrand[..]
...")
        .run();
```
`...` is a multi-line glob.  `str![]` isn't as important in these cases as we can't really update snapshots like these.